### PR TITLE
CC36

### PR DESCRIPTION
**Pull request by nojadzworld**
- [x] Top-level README “Table of Contents” is updated
 - [x] README for this challenge is complete
       - [x] Summary, Description, Approach & Efficiency, Solution
       - [x] Picture of whiteboard
       - [x] Link to code
 - [x] Feature tasks for this challenge are completed
 - [x] Unit tests written and passing
       - [x] “Happy Path” - Expected outcome
       - [x] Expected failure
       - [x] Edge Case (if applicable/obvious)

### CODE REVIEW

File: java/datastructures/lib/src/test/java/datastructures/graph/GraphTest.java [hunk 1]
```diff
@@ -81,4 +82,39 @@ public void testSingleNodeAndEdgeGraph() {
...
+        graph.addVertex("A");
+        List<Vertex<String>> result = graph.breadthFirst();
+        assertEquals(1, result.size());
+        assertEquals("A", result.get(0).value);
+    }
+
+    @Test
+    public void testBreadthFirstMultiple() {
+        Graph<Character> graph = new Graph<>(1  );
+        graph.addEdge(new Vertex<>('A'), new Vertex<>('B'));
```
Comment (lucy-gelderloos) [inline java/datastructures/lib/src/test/java/datastructures/graph/GraphTest.java:105]: These vertices are getting created, but not added to the graph, so the test throws an error. The clunky way to do this is to manually create each vertex, add it to the graph, and then pass the vertices into your `addEdge` method. 

I bet you could also do something cool with `addEdge`, where if you pass in vertices that don't exist, it will call `addVertex` to create those as well.
